Tests/TrackerTests.m

```diff
@@ -118,6 +119,116 @@
+
+                [tracker sendResponseHeaders:@{@"foo": @"bar"} trackingIdentifier:100];
+            });
+
+        });
+
+        context(@"perfData", ^{
+            __block _RPTTracker *tracker;
+            beforeEach(^{
+                _RPTMeasurement *measurement = [[_RPTMeasurement alloc] init];
+                measurement.trackingIdentifier = 100;
+                measurement.receiver = @"https://google.com";
+                _RPTRingBuffer* buffer = [_RPTRingBuffer nullMock];
+                [buffer stub:@selector(measurementWithTrackingIdentifier:) andReturn:measurement];
+                tracker = [_RPTTracker.alloc initWithRingBuffer:buffer currentMetric:nil];
+            });
+
+            afterEach(^{
+                tracker = nil;
+            });
+
+            it(@"should call 'sendEventName:topLevelDataObject:' method with the eventName is 'perf' ", ^{
+                KWCaptureSpy *spy = [_RPTEventBroadcast captureArgument:@selector(sendEventName:topLevelDataObject:) atIndex:0];
+
+                [tracker sendResponseHeaders:@{@"foo": @"bar"} trackingIdentifier:100];
+
+                NSString *eventName = spy.argument;
+                [[eventName should] equal:@"perf"];
+            });
+
+            it(@"should call 'sendEventName:topLevelDataObject:' method with the topLevelDataObject.perfData.type is 'resource' ", ^{
+                KWCaptureSpy *spy = [_RPTEventBroadcast captureArgument:@selector(sendEventName:topLevelDataObject:) atIndex:1];
+
+                [tracker sendResponseHeaders:@{@"foo": @"bar"} trackingIdentifier:100];
+
+                NSDictionary *dict = spy.argument;
+                [[dict[@"perfdata"][@"type"] should] equal:@"resource"];
+            });
+
+            it(@"should call 'sendEventName:topLevelDataObject:' method with the topLevelDataObject.perfData.entries having one item", ^{
+                KWCaptureSpy *spy = [_RPTEventBroadcast captureArgument:@selector(sendEventName:topLevelDataObject:) atIndex:1];
+
+                [tracker sendResponseHeaders:@{@"foo": @"bar"} trackingIdentifier:100];
+
+                NSDictionary *dict = spy.argument;
+                NSArray *entries = dict[@"perfdata"][@"entries"];
+                [[[entries should] have:1] items];
+            });
+
+            it(@"should call 'sendEventName:topLevelDataObject:' method with the topLevelDataObject.perfData.entries having one item which have 'name' is equal to the url of measurement", ^{
+                KWCaptureSpy *spy = [_RPTEventBroadcast captureArgument:@selector(sendEventName:topLevelDataObject:) atIndex:1];
+
+                [tracker sendResponseHeaders:@{@"foo": @"bar"} trackingIdentifier:100];
+
+                NSDictionary *dict = spy.argument;
+                NSArray *entries = dict[@"perfdata"][@"entries"];
+                NSDictionary *entryData = entries.firstObject;
+                [[entryData[@"name"] should] equal:@"https://google.com"];
```

Comment:
Why is the url `https://google.com`? I know why, but it isn't obvious from reading the test, so I think setting the `measurement.receiver` should be in the actual test not the beforeEach.

Reply:
I will update it.